malasakit-django/pcari/admin.py

```diff
@@ -1,313 +1,177 @@
+
+    # Sets fields as readonly
+    readonly_fields = ('respondent', 'question', 'language', 'message')
+
+    # Enables search
+    search_fields = ('message', 'tag')
+
+
+@admin.register(QuantitativeQuestionRating)
+class QuantitativeQuestionRatingAdmin(ResponseAdmin):
+    """
+    Customizes admin change page functionality for QuantitativeQuestionRatings
+    """
+    def get_question_prompt(self, obj):
+        """
+        A callable that returns the `prompt` attribute of the ForeignKey `question`.
+        """
+        return obj.question.prompt
+
+    # Columns to display in the Comment change list page, in order from left to right 
+    list_display = ('respondent', 'get_question_prompt', 'timestamp', 'score')
+
+    # By default first column listed in list_display is clickable; this makes `message` column clickable
+    list_display_links = ('get_question_prompt',)
+
+    # Specify which columns we want filtering capabilities for
+    list_filter = ('timestamp', 'score')
+
+    # Sets fields as readonly
+    readonly_fields = ('respondent', 'question', 'timestamp', 'score')
+
+    # Enables search
+    search_fields = ('question__prompt', 'score')
+
+
+class QuestionAdmin(admin.ModelAdmin):
+    """
+    Abstract admin class for QualitativeQuestionAdmin and QuantitativeQuestionAdmin
+    """
+    # Performance optimizer to limit database queries
+    list_select_related = True
+
+    # Adds a "Save as New" button
+    save_as = True
+        
+
+@admin.register(QualitativeQuestion)
+class QualitativeQuestionAdmin(QuestionAdmin):
+    """
+    Customizes admin change page functionality for QualitativeQuestionAdmin
+    """
+    def get_comments(self, obj):
+        return list(obj.comments)
+
+    # Columns to display in the Comment change list page, in order from left to right 
+    list_display = ('prompt', 'tag', 'get_comments')
+
+    # Specify which columns we want filtering capabilities for
+    list_filter = ('tag', 'prompt')
+
+    # Enables search
+    search_fields = ('prompt', 'tag')
+
+
+@admin.register(QuantitativeQuestion)
+class QuantitativeQuestionAdmin(QuestionAdmin):
+    """
+    Customizes admin change page functionality for QuantitativeQuestionAdmin
+    """
+    # Columns to display in the Comment change list page, in order from left to right
+    list_display = ('prompt', 'tag')
+
+    # Specify which columns we want filtering capabilities for
+    list_filter = ('tag',)
+
+    # Sets fields as readonly
+    readonly_fields = ('prompt', 'tag')
+
+    # Enables search
+    search_fields = ('prompt', 'tag')
+
+
+@admin.register(Respondent)
+class RespondentAdmin(admin.ModelAdmin):
+    """
+    Customizes admin change page functionality for RespondentAdmin
+    """
+    def get_comments(self, obj):
+        return list(obj.comments_made)
+
+    # Empty responses (recorded as None) will be replaced by this placeholder
+    empty_value_display = '-- empty response --'
```

Comment:
Because `age` and `gender` are narrow columns, this value doesn't wrap well. Maybe something like `(Empty)`?